### ISSUE TITLE
Filters with multiple verticals break when switching verticals

### ISSUE THREAD

**Issue by Jeroen00** [microsoft-search/pnp-modern-search#2818]
**Version used**
4.8.0

**Describe the bug**
We are building a search page with the verticals web part with multiple tabs each connected to one search results web part. We have one filter web part connected to the verticals and all the search results web parts. Filtering works but when we have an active filter and we switch to a vertical with no results and then switch back to a vertical with results, the active filter checkboxes become unselected and the Apply and Clear buttons are disabled.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Create a page with a search box, verticals with 2 or more tabs and linked search results web parts
2. Start a search on the main tab which returns results
3. Apply one filter
4. Switch to a tab with 0 results
5. Switch back to a tab with > 0 results

**Expected behavior**
Filter checkboxes and buttons remain working when switching back from a vertical tab with 0 results

**Screenshots**
  
**Initial search**
![image](https://user-images.githubusercontent.com/42589264/222390368-0cf0bdee-161e-4752-a103-ff56a582d01a.png)
  
  
**Switch to tab with 0 results**
![image](https://user-images.githubusercontent.com/42589264/222390446-52839d48-1ce2-4a0a-a22e-464bd285e6e7.png)
  
  
**Switch to tab with >0 results**
![image](https://user-images.githubusercontent.com/42589264/222390540-1c985ff5-e75c-498f-b411-f4beebb48e6b.png)
  
  
**Desktop (please complete the following information):**
 - Browser: Chrome

**Additional context**
None

**Comment by Jeroen00** [comment]
hi @wobba Do you maybe have some input on this issue? Thanks!

**Comment by wobba** [comment]
No, someone has to take the time to triage and spend time to fix it, which is why the issue is tagged this way :)

**Comment by ValerasNarbutas** [comment]
@wobba  i could take a look

**Comment by wobba** [comment]
@ValerasNarbutas feel free :)
